src/ruby/puppetserver-lib/puppet/server/execution.rb

```diff
@@ -7,31 +7,41 @@ class Puppet::Server::Execution
   def self.initialize_execution_stub
     Puppet::Util::ExecutionStub.set do |command, options, stdin, stdout, stderr|
       if command.is_a?(Array)
+        orig_command_str = command.join(" ")
         binary = command.first
         args = command[1..-1]
       else
+        orig_command_str = command
         binary = command
         args = nil
       end
 
-      # TODO - options is currently ignored - https://tickets.puppetlabs.com/browse/SERVER-74
+      # TODO - options relating to stdout/stderr are not yet handled, see SERVER-74 and
+      #  PUP-6640
 
       # We're going to handle STDIN/STDOUT/STDERR in java, so we don't need
       # them here.  However, Puppet::Util::Execution.execute doesn't close them
       # for us, so we have to do that now.
       [stdin, stdout, stderr].each { |io| io.close rescue nil }
 
-      execute(binary, args)
+      execute(orig_command_str, binary, options, args)
     end
   end
 
-  def self.execute(command, args = nil)
+  private
+  def self.execute(orig_command_str, command, options, args)
```

Comment:
I waffled on this too, and mostly did it this way because the logic was already here and was already splitting up the command, and I didn't want to have to re-assemble the command string inside of the private method.  But I think what you're suggesting makes sense, will take a quick pass to see what it looks like if I just push *all* of that logic down into the private method.